sdks/the-compact-sdk/src/builders/claim.ts

```diff
@@ -0,0 +1,1521 @@
+   * @param expires - Unix timestamp when this claim expires
+   * @returns This builder for chaining
+   */
+  expires(expires: bigint): this {
+    this._expires = expires
+    return this
+  }
+
+  /**
+   * Set allocator-specific data
+   * This is typically the allocator's signature authorizing the claim
+   * @param data - Allocator data (usually a signature)
+   * @returns This builder for chaining
+   */
+  allocatorData(data: `0x${string}`): this {
+    this._allocatorData = data
+    return this
+  }
+
+  /**
+   * Set the total amount allocated from the resource lock for this claim
+   * This is the sum of all claimant amounts
+   * @param amount - Total allocated amount in wei
+   * @returns This builder for chaining
+   */
+  allocatedAmount(amount: bigint): this {
+    this._allocatedAmount = amount
+    return this
+  }
+
+  /**
+   * Set the resource lock ID being claimed
+   * This identifies which specific lock the claim is processing
+   * @param id - The lock ID (ERC6909 token ID)
+   * @returns This builder for chaining
+   */
+  id(id: bigint): this {
+    this._id = id
+    return this
+  }
+
+  /**
+   * Set the lock tag for building claimants
+   * Must be set before adding any claimants (transfers, converts, withdrawals)
+   * @param lockTag - 12-byte lock tag identifying allocator, scope, and reset period
+   * @returns This builder for chaining
+   */
+  lockTag(lockTag: `0x${string}`): this {
+    this._lockTag = lockTag
+    return this
+  }
+
+  /**
+   * Set the sponsor's signature on the compact
+   * Use empty string ('') for registered claims where hash was pre-registered on-chain
+   * @param signature - Sponsor's EIP-712 signature or empty string for registered claims
+   * @returns This builder for chaining
+   */
+  sponsorSignature(signature: `0x${string}` | ''): this {
+    this._sponsorSignature = signature as `0x${string}`
+    return this
+  }
+
+  /**
+   * Set witness mandate data for compact verification
+   * The witness provides additional authorization context for the claim
+   * @param mandateType - The mandate type definition with hash and typestring
+   * @param mandate - The mandate data object
+   * @returns This builder for chaining
+   */
+  witness<T extends object>(mandateType: MandateType<T>, mandate: T): this {
+    this._witness = mandateType.hash(mandate)
+    this._witnessTypestring = mandateType.witnessTypestring
+    return this
+  }
+
+  /**
+   * Add a transfer claimant that receives ERC6909 tokens with the same lock tag
+   * Tokens remain within The Compact as ERC6909s
+   * @param params - Transfer parameters
+   * @param params.recipient - Address to receive the transferred tokens
+   * @param params.amount - Amount to transfer in wei
+   * @returns This builder for chaining
+   * @throws {Error} If lockTag is not set before calling this method
+   */
+  addTransfer(params: { recipient: `0x${string}`; amount: bigint }): this {
```

Comment:
probably worth doing some isAddress validation with viem or ethers just to check eip55 checksum and length etc. Presume there should be better types than 0xstring also